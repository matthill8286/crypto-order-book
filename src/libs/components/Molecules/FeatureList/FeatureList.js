import React from 'react';
import { Divider } from "../../Atoms/Divider";
import { CopyText } from "../../Atoms/Typography";
import { FlexBox } from "../../Helper/FlexBox";
import { Cell, Grid, Row } from "../../Helper/Grid";
import { styled } from "../../../styles";
export var ListWrapper = styled.div.withConfig({
  displayName: "FeatureList__ListWrapper",
  componentId: "sc-1lbwj8e-0"
})(["padding-top:", ";padding-bottom:", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.spacing.base.sm;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.spacing.base.sm;
});
export var FeatureList = function FeatureList(_ref3) {
  var list = _ref3.list;

  if (!list || !Array.isArray(list) || !list.length) {
    return null;
  }

  return React.createElement(FlexBox, {
    display: "flex",
    flexDirection: "column"
  }, list.map(function (item, idx) {
    return item.label !== '' && item.label ? React.createElement(React.Fragment, {
      key: "FeatureList-".concat(item.label, "-").concat(idx)
    }, React.createElement(ListWrapper, null, React.createElement(Grid, {
      noPadding: true
    }, React.createElement(Row, {
      noMargin: true
    }, React.createElement(Cell, {
      columns: 6,
      colsMd: 4,
      colsSm: 4,
      colsXs: 2
    }, React.createElement(CopyText, {
      weight: "semibold",
      color: "grey5",
      margin: "0"
    }, item.label)), React.createElement(Cell, {
      columns: 6,
      colsMd: 4,
      colsSm: 4,
      colsXs: 2
    }, React.createElement(CopyText, {
      color: "grey5",
      margin: "0"
    }, item.value))))), React.createElement(Divider, {
      color: "grey2",
      height: 1
    })) : null;
  }));
};
FeatureList.displayName = 'FeatureList';
//# sourceMappingURL=FeatureList.js.map