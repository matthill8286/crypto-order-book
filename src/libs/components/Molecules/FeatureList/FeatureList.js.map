{"version":3,"sources":["../../../../src/components/Molecules/FeatureList/FeatureList.tsx"],"names":["React","Divider","CopyText","FlexBox","Cell","Grid","Row","styled","ListWrapper","div","theme","spacing","base","sm","FeatureList","list","Array","isArray","length","map","item","idx","label","value","displayName"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,OAAT;AACA,SAASC,QAAT;AACA,SAASC,OAAT;AACA,SAASC,IAAT,EAAeC,IAAf,EAAqBC,GAArB;AACA,SAASC,MAAT;AAQA,OAAO,IAAMC,WAAW,GAAGD,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,8CACP;AAAA,MAAGC,KAAH,QAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,OAAN,CAAcC,IAAd,CAAmBC,EAAlC;AAAA,CADO,EAEJ;AAAA,MAAGH,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,OAAN,CAAcC,IAAd,CAAmBC,EAAlC;AAAA,CAFI,CAAjB;AAKP,OAAO,IAAMC,WAAuC,GAAG,SAA1CA,WAA0C,QAAkC;AAAA,MAA/BC,IAA+B,SAA/BA,IAA+B;;AACvF,MAAI,CAACA,IAAD,IAAS,CAACC,KAAK,CAACC,OAAN,CAAcF,IAAd,CAAV,IAAiC,CAACA,IAAI,CAACG,MAA3C,EAAmD;AACjD,WAAO,IAAP;AACD;;AAED,SACE,oBAAC,OAAD;AAAS,IAAA,OAAO,EAAC,MAAjB;AAAwB,IAAA,aAAa,EAAC;AAAtC,KACGH,IAAI,CAACI,GAAL,CAAS,UAACC,IAAD,EAAOC,GAAP;AAAA,WACRD,IAAI,CAACE,KAAL,KAAe,EAAf,IAAqBF,IAAI,CAACE,KAA1B,GACE,oBAAC,KAAD,CAAO,QAAP;AAAgB,MAAA,GAAG,wBAAiBF,IAAI,CAACE,KAAtB,cAA+BD,GAA/B;AAAnB,OACE,oBAAC,WAAD,QACE,oBAAC,IAAD;AAAM,MAAA,SAAS;AAAf,OACE,oBAAC,GAAD;AAAK,MAAA,QAAQ;AAAb,OACE,oBAAC,IAAD;AAAM,MAAA,OAAO,EAAE,CAAf;AAAkB,MAAA,MAAM,EAAE,CAA1B;AAA6B,MAAA,MAAM,EAAE,CAArC;AAAwC,MAAA,MAAM,EAAE;AAAhD,OACE,oBAAC,QAAD;AAAU,MAAA,MAAM,EAAC,UAAjB;AAA4B,MAAA,KAAK,EAAC,OAAlC;AAA0C,MAAA,MAAM,EAAC;AAAjD,OACGD,IAAI,CAACE,KADR,CADF,CADF,EAME,oBAAC,IAAD;AAAM,MAAA,OAAO,EAAE,CAAf;AAAkB,MAAA,MAAM,EAAE,CAA1B;AAA6B,MAAA,MAAM,EAAE,CAArC;AAAwC,MAAA,MAAM,EAAE;AAAhD,OACE,oBAAC,QAAD;AAAU,MAAA,KAAK,EAAC,OAAhB;AAAwB,MAAA,MAAM,EAAC;AAA/B,OACGF,IAAI,CAACG,KADR,CADF,CANF,CADF,CADF,CADF,EAiBE,oBAAC,OAAD;AAAS,MAAA,KAAK,EAAC,OAAf;AAAuB,MAAA,MAAM,EAAE;AAA/B,MAjBF,CADF,GAoBI,IArBI;AAAA,GAAT,CADH,CADF;AA2BD,CAhCM;AAkCPT,WAAW,CAACU,WAAZ,GAA0B,aAA1B","sourcesContent":["import React from 'react'\nimport { Divider } from '@/components/Atoms/Divider'\nimport { CopyText } from '@/components/Atoms/Typography'\nimport { FlexBox } from '@/components/Helper/FlexBox'\nimport { Cell, Grid, Row } from '@/components/Helper/Grid'\nimport { styled } from '@/styles'\n\nexport type FeatureListType = { label?: string; value: string }\n\nexport interface FeatureListProps {\n  list: FeatureListType[]\n}\n\nexport const ListWrapper = styled.div`\n  padding-top: ${({ theme }) => theme.spacing.base.sm};\n  padding-bottom: ${({ theme }) => theme.spacing.base.sm};\n`\n\nexport const FeatureList: React.FC<FeatureListProps> = ({ list }): JSX.Element | null => {\n  if (!list || !Array.isArray(list) || !list.length) {\n    return null\n  }\n\n  return (\n    <FlexBox display=\"flex\" flexDirection=\"column\">\n      {list.map((item, idx) =>\n        item.label !== '' && item.label ? (\n          <React.Fragment key={`FeatureList-${item.label}-${idx}`}>\n            <ListWrapper>\n              <Grid noPadding>\n                <Row noMargin>\n                  <Cell columns={6} colsMd={4} colsSm={4} colsXs={2}>\n                    <CopyText weight=\"semibold\" color=\"grey5\" margin=\"0\">\n                      {item.label}\n                    </CopyText>\n                  </Cell>\n                  <Cell columns={6} colsMd={4} colsSm={4} colsXs={2}>\n                    <CopyText color=\"grey5\" margin=\"0\">\n                      {item.value}\n                    </CopyText>\n                  </Cell>\n                </Row>\n              </Grid>\n            </ListWrapper>\n            <Divider color=\"grey2\" height={1} />\n          </React.Fragment>\n        ) : null\n      )}\n    </FlexBox>\n  )\n}\n\nFeatureList.displayName = 'FeatureList'\n"],"file":"FeatureList.js"}