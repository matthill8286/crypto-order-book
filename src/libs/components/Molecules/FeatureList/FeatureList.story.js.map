{"version":3,"sources":["../../../../src/components/Molecules/FeatureList/FeatureList.story.tsx"],"names":["storiesOf","text","React","FeatureList","module","add","width","label","value"],"mappings":"AAAA,SAASA,SAAT,QAA0B,kBAA1B;AACA,SAASC,IAAT,QAAqB,wBAArB;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEAH,SAAS,CAAC,qCAAD,EAAwCI,MAAxC,CAAT,CAAyDC,GAAzD,CAA6D,SAA7D,EAAwE;AAAA,SACtE;AAAK,IAAA,KAAK,EAAE;AAAEC,MAAAA,KAAK,EAAE;AAAT;AAAZ,KACE,oBAAC,WAAD;AACE,IAAA,IAAI,EAAE,CACJ;AAAEC,MAAAA,KAAK,EAAEN,IAAI,CAAC,UAAD,EAAa,UAAb,CAAb;AAAuCO,MAAAA,KAAK,EAAE;AAA9C,KADI,EAEJ;AAAED,MAAAA,KAAK,EAAEN,IAAI,CAAC,oBAAD,EAAuB,oBAAvB,CAAb;AAA2DO,MAAAA,KAAK,EAAE;AAAlE,KAFI,EAGJ;AAAED,MAAAA,KAAK,EAAEN,IAAI,CAAC,MAAD,EAAS,MAAT,CAAb;AAA+BO,MAAAA,KAAK,EAAE;AAAtC,KAHI,EAIJ;AAAED,MAAAA,KAAK,EAAEN,IAAI,CAAC,QAAD,EAAW,QAAX,CAAb;AAAmCO,MAAAA,KAAK,EAAE;AAA1C,KAJI;AADR,IADF,CADsE;AAAA,CAAxE","sourcesContent":["import { storiesOf } from '@storybook/react'\nimport { text } from '@storybook/addon-knobs'\nimport * as React from 'react'\nimport { FeatureList } from './FeatureList'\n\nstoriesOf('Design System/Molecules/FeatureList', module).add('Default', () => (\n  <div style={{ width: '400px' }}>\n    <FeatureList\n      list={[\n        { label: text('Provider', 'Provider'), value: 'Youtube' },\n        { label: text('Primary competency', 'Primary competency'), value: 'Digital Marketing' },\n        { label: text('Type', 'Type'), value: 'Article' },\n        { label: text('Length', 'Length'), value: '16 minutes' },\n      ]}\n    />\n  </div>\n))\n"],"file":"FeatureList.story.js"}