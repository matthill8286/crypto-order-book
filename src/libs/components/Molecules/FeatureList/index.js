export * from "./FeatureList";
//# sourceMappingURL=index.js.map