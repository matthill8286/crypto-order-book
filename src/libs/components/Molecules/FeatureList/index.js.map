{"version":3,"sources":["../../../../src/components/Molecules/FeatureList/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './FeatureList'\n"],"file":"index.js"}