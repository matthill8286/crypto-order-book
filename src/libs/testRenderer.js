import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map