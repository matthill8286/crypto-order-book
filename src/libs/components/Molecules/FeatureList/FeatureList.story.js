import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import * as React from 'react';
import { FeatureList } from "./FeatureList";
storiesOf('Design System/Molecules/FeatureList', module).add('Default', function () {
  return React.createElement("div", {
    style: {
      width: '400px'
    }
  }, React.createElement(FeatureList, {
    list: [{
      label: text('Provider', 'Provider'),
      value: 'Youtube'
    }, {
      label: text('Primary competency', 'Primary competency'),
      value: 'Digital Marketing'
    }, {
      label: text('Type', 'Type'),
      value: 'Article'
    }, {
      label: text('Length', 'Length'),
      value: '16 minutes'
    }]
  }));
});
//# sourceMappingURL=FeatureList.story.js.map