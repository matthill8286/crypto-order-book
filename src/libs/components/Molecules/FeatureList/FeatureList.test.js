import React from 'react';
import { mountWithTheme, renderWithTheme } from "../../../testRenderer";
import { FeatureList, ListWrapper } from "./FeatureList";
describe('Meta Feature list', function () {
  var inputProps;
  beforeEach(function () {
    inputProps = [{
      label: 'Provider',
      value: 'Youtube'
    }, {
      label: 'Primary competency',
      value: 'Digital Marketing'
    }, {
      label: 'Type',
      value: 'Article'
    }, {
      label: 'Length',
      value: '16 minutes'
    }];
  });
  it('renders feature list', function () {
    var wrapper = renderWithTheme(React.createElement(FeatureList, {
      list: inputProps
    }));
    expect(wrapper).toMatchSnapshot();
  });
  it('renders feature list with 3 children', function () {
    var component = mountWithTheme(React.createElement(FeatureList, {
      list: inputProps.slice(0, 3)
    }));
    var find = component.find(ListWrapper);
    expect(find.length).toBe(3);
  });
  it('removes the entry with no valid label', function () {
    var listWithMissingLabel = [{
      label: '',
      value: 'Youtube'
    }, {
      label: 'Type',
      value: 'Article'
    }, {
      label: 'Primary competency',
      value: 'Digital Marketing'
    }, {
      label: 'Type',
      value: 'Article'
    }];
    var component = mountWithTheme(React.createElement(FeatureList, {
      list: listWithMissingLabel
    }));
    var child = component.childAt(0);
    var childProps = child.props().list;
    expect(childProps).toStrictEqual(listWithMissingLabel);
    expect(childProps.length).toBe(4);
    var find = component.find(ListWrapper);
    expect(find.length).toBe(3);
  });
});
//# sourceMappingURL=FeatureList.test.js.map