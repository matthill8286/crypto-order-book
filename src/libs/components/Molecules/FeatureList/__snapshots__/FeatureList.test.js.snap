// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta Feature list renders feature list 1`] = `
<div
  className="FlexBox__StyledBox-sc-52meld-0 kSmAcZ"
  display="flex"
>
  <div
    className="FeatureList__ListWrapper-sc-1lbwj8e-0 jxsuMD"
  >
    <div
      className="Grid__StyledGrid-ovutvr-0 bFTtP"
    >
      <div
        className="Row__StyledRow-sc-1o9ecac-0 dDvLOq"
      >
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWfVXO"
            color="grey5"
            fontSize="xs"
          >
            Provider
          </p>
        </div>
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 gMqzZI"
            color="grey5"
            fontSize="xs"
          >
            Youtube
          </p>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="Divider__StyledHR-sc-40hr1c-0 gxbmei"
    color="grey2"
    height={1}
  />
  <div
    className="FeatureList__ListWrapper-sc-1lbwj8e-0 jxsuMD"
  >
    <div
      className="Grid__StyledGrid-ovutvr-0 bFTtP"
    >
      <div
        className="Row__StyledRow-sc-1o9ecac-0 dDvLOq"
      >
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWfVXO"
            color="grey5"
            fontSize="xs"
          >
            Primary competency
          </p>
        </div>
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 gMqzZI"
            color="grey5"
            fontSize="xs"
          >
            Digital Marketing
          </p>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="Divider__StyledHR-sc-40hr1c-0 gxbmei"
    color="grey2"
    height={1}
  />
  <div
    className="FeatureList__ListWrapper-sc-1lbwj8e-0 jxsuMD"
  >
    <div
      className="Grid__StyledGrid-ovutvr-0 bFTtP"
    >
      <div
        className="Row__StyledRow-sc-1o9ecac-0 dDvLOq"
      >
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWfVXO"
            color="grey5"
            fontSize="xs"
          >
            Type
          </p>
        </div>
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 gMqzZI"
            color="grey5"
            fontSize="xs"
          >
            Article
          </p>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="Divider__StyledHR-sc-40hr1c-0 gxbmei"
    color="grey2"
    height={1}
  />
  <div
    className="FeatureList__ListWrapper-sc-1lbwj8e-0 jxsuMD"
  >
    <div
      className="Grid__StyledGrid-ovutvr-0 bFTtP"
    >
      <div
        className="Row__StyledRow-sc-1o9ecac-0 dDvLOq"
      >
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWfVXO"
            color="grey5"
            fontSize="xs"
          >
            Length
          </p>
        </div>
        <div
          className="Cellstyled__StyledCell-sc-4z19ev-0 kGEQEJ"
        >
          <p
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 gMqzZI"
            color="grey5"
            fontSize="xs"
          >
            16 minutes
          </p>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="Divider__StyledHR-sc-40hr1c-0 gxbmei"
    color="grey2"
    height={1}
  />
</div>
`;
